src/include/compat.h

```diff
@@ -110,4 +112,25 @@
 #define MSG_DONTWAIT MSG_NONBLOCK
 #endif
 
+#if defined(HAVE_PTHREAD_SETNAME_NP)
+#define PTHREAD_SETNAME_NP(thread, name) pthread_setname_np(thread, name)
```

Comment:
On 24-6-2016 12:57, Kefu Chai wrote:
> In src/include/compat.h
> <https://github.com/ceph/ceph/pull/9763#discussion_r68383203>:
> 
>> @@ -110,4 +112,25 @@
>>  #define MSG_DONTWAIT MSG_NONBLOCK
>>  #endif
>>  
>> +#if defined(HAVE_PTHREAD_SETNAME_NP)
>> +#define PTHREAD_SETNAME_NP(thread, name) pthread_setname_np(thread, name)
> 
> much better. imo, can we avoid having the all upper case function name,
> i.e. reuse the name |pthread_setname_np()|, and define it with its
> equivalent if it's not available? i know, people might want to argue
> that macro should be named as |FUNCTION()| instead of |function()|.

I'm sort of OLD school here, and I'd prefer MACROs and DEFINES to be
prefably in CAPS. Especially to make a distinction with code wrapped in
classed.

Other solution that was used in the time classes is the prefix the
repair code with ceph_. So there is a signal that something is up.

The advantage of reusing the name that we do not have to educate other
devs to think/look seriously at their code.

I'm all for education. :)